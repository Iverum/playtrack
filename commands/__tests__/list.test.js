/* eslint-disable no-undef */
const Database = require('../../db');
const list = require('../list');

let db = null;
beforeEach(async () => {
  db = new Database(':memory:');
  await db.initialize();
});

test('fails when no db is provided', async () => {
  const response = await list(null);
  expect(response).toEqual('Could not find database.');
});

test('fails when no games are found', async () => {
  const response = await list(db);
  expect(response).toEqual('No games found.');
});

test('returns game results', async () => {
  await db.run(`
    INSERT INTO games (name, completed)
    VALUES ("Metal Gear", 0), ("Metal Gear 2", 1);
  `);
  const response = await list(db);
  expect(response).toMatchSnapshot();
});
