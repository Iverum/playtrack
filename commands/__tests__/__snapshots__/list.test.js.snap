// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns game results 1`] = `
"NAME         FIRST PLAYED        LAST PLAYED        
Metal Gear   2021-01-02 00:23:34 2021-01-02 00:23:34
Metal Gear 2 2021-01-02 00:23:34 2021-01-02 00:23:34"
`;
